iotlabcli/parser/main.py

```diff
@@ -0,0 +1,113 @@
+# modify and/ or redistribute the software under the terms of the CeCILL
+# license as circulated by CEA, CNRS and INRIA at the following URL
+# http://www.cecill.info.
+#
+# As a counterpart to the access to the source code and  rights to copy,
+# modify and redistribute granted by the license, users are provided only
+# with a limited warranty  and the software's author,  the holder of the
+# economic rights,  and the successive licensors  have only  limited
+# liability.
+#
+# The fact that you are presently reading this means that you have had
+# knowledge of the CeCILL license and that you accept its terms.
+
+"""Main parser."""
+
+import sys
+from argparse import ArgumentParser
+
+import iotlabcli.parser.auth
+import iotlabcli.parser.admin
+import iotlabcli.parser.experiment
+import iotlabcli.parser.node
+import iotlabcli.parser.profile
+import iotlabcli.parser.robot
+
+# from aggregation-tools
+try:
+    from iotlabaggregator.serial import main as aggregator_serial
+    from iotlabaggregator.sniffer import main as aggregator_sniffer
+    AGGREGATION_TOOLS = True
+except ImportError:
+    AGGREGATION_TOOLS = False
+
+# from ssh-cli-tools
+try:
+    from iotlabsshcli.parser.open_a8_parser import main as ssh
+    SSH_TOOLS = True
+except ImportError:
+    SSH_TOOLS = False
+
+# from oml-plot-tools
+try:
+    from oml_plot_tools.consum import main as oml_plot_consum
+    from oml_plot_tools.radio import main as oml_plot_radio
+    from oml_plot_tools.traj import main as oml_plot_traj
+    OMLPLOT_TOOLS = True
+except ImportError:
+    OMLPLOT_TOOLS = False
+
+
+def parse_subcommands(commands, args=None):
+    """ common function to parse `iotlab` or other with subcommands """
+    args = args or sys.argv[1:]
+
+    parser = ArgumentParser()
+    parser.add_argument('command', nargs='?',
+                        choices=commands.keys(), default='help')
+    commands['help'] = lambda args: parser.print_help()
+
+    opts, args = parser.parse_known_args(args)
+
+    return commands[opts.command](args)
+
+
+def aggregator(args):
+    """'iotlab aggregator' main function."""
+
+    commands = {
+        'sniffer': aggregator_serial,
+        'serial': aggregator_sniffer
+    }
+    parse_subcommands(commands, args)
+
+
+def oml_plot(args):
+    """'iotlab aggregator' main function."""
+
+    commands = {
+        'consum': oml_plot_consum,
+        'radio': oml_plot_radio,
+        'traj': oml_plot_traj
+    }
+    parse_subcommands(commands, args)
+
+
+def main(args=None):
+    """'iotlab' main function."""
+
+    commands = {
+        'auth': iotlabcli.parser.auth.main,
+        'admin': iotlabcli.parser.admin.main,
+        'experiment': iotlabcli.parser.experiment.main,
+        'node': iotlabcli.parser.node.main,
+        'profile': iotlabcli.parser.profile.main,
+        'robot': iotlabcli.parser.robot.main,
+        'help': None
+    }
+    if AGGREGATION_TOOLS:
+        commands['aggregator'] = aggregator
```

Comment:
Instead of this, I would directly map `serial` and `sniffer` subcommands. Otherwise, this ends with a command like `iotlab aggregator serial ...`. Using `iotlab serial ...` would be simpler. The same applies to `oml_plot_tools`.
If this is accepted, the `aggregator` and `oml_plot` functions can simply be removed.

Reply:
Yes, `iotlab serial`; `iotlab sniffer` would be nice, (though it obscures the fact they are aggregators) but I was cautious not to introduce too many top-level subcommands. To recap:
```
iotlab serial
iotlab sniffer
iotlab traj
iotlab consum
iotlab radio
````
these don't clash with existing subcommands, but it's also a lot to add. oml_plot_tools commands are relatively niche I think so leaving them under `iotlab oml-plot` seems better. We need a third opinion I guess

Reply:
+1 for discussing this with others